distage-example-monomorphic-cats/src/main/scala/leaderboard/repo/Ladder.scala

```diff
@@ -0,0 +1,63 @@
+package leaderboard.repo
+
+import cats.effect.IO
+import cats.implicits.*
+import distage.Lifecycle
+import doobie.postgres.implicits.*
+import doobie.syntax.string.*
+import leaderboard.model.{QueryFailure, Score, UserId}
+import leaderboard.sql.SQL
+import logstage.LogIO
+
+import scala.collection.concurrent.TrieMap
+
+trait Ladder {
+  def submitScore(userId: UserId, score: Score): IO[Either[QueryFailure, Unit]]
```

Comment:
Use `IO.raiseError` / `F.raiseError` for monofunctor examples. No need to try to imitate typed errors with Either.
```suggestion
  def submitScore(userId: UserId, score: Score): IO[Unit]
```